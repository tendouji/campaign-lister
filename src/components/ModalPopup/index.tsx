import React, { Component } from 'react';
import styled from 'styled-components';
import { colors, layout, styles } from "../../constants/ui";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTimes } from "@fortawesome/free-solid-svg-icons/faTimes";


type ModalPopupProps = {
    show: boolean,
    title: string,
    content: string,
    close?: () => void
}

class ModalPopup extends Component<ModalPopupProps, {}> {
    constructor(props: ModalPopupProps) {
        super(props);
        this.closeHandler = this.closeHandler.bind(this);
    }

    closeHandler = () => {
        const { close } = this.props;
        if(!!close) close();
    };

    render() {
        const {
            show,
            title,
            content
        } = this.props;

        return (
            <ModalPopupWrapper
                className={[
                    'modal-popup',
                    !!show ? ` show` : '',
                ].join('')}
            >
                <div className="bg" />
                <div className="fg">
                    <div className="header">
                        <div className="title">{title}</div>
                        <div className="close" onClick={this.closeHandler}>
                            <FontAwesomeIcon icon={faTimes} size={'lg'} />
                        </div>
                    </div>
                    <div className="content">
                        <div dangerouslySetInnerHTML={{__html: content}} />
                    </div>
                </div>
            </ModalPopupWrapper>
        )
    }
}

export default ModalPopup;


const closeSize = '3rem';
const ModalPopupWrapper = styled.div`
    display: none;
    position: fixed;
    top: 0;
    right: 0;
    width: 100%;
    height: 100%;
    justify-content: center;
    align-items: center;
    z-index: 5;
    
    & > .bg {
        position: absolute;
        top: 0;
        right: 0;
        width: 100%;
        height: 100%;
        background-color: rgba(255, 255, 255, 0.8);
    }
    
    &.show {
        display: flex;
    }
    
    & > .fg {
        display: inline-block;
        position: relative;
        width: calc(100% - 2 * ${layout.standardGap});
        height: calc(100% - 2 * ${layout.standardGap});
        border: ${styles.lineStyle};
        border-radius: ${layout.borderRadius};
        box-sizing: border-box;
        background-color: ${colors.white};
        
        & > .header {
            display: flex;
            height: ${closeSize};
            border-bottom: ${styles.lineStyle};
            justify-content: space-between;
            align-items: center;
            
            & .title {
                width: calc(100% - ${closeSize});
                padding: 0 ${layout.standardGap} ;
                box-sizing: border-box;
                font-weight: 700;
                ${styles.textOverflowStyle}
            }
            
            & .close {
                ${styles.filterBarButtonStyle};
                display: flex;
                height: ${closeSize};
                width: ${closeSize};
                font-size: ${layout.fontSizeLarge};
                justify-content: center;
                align-items: center;
                cursor: pointer;
            }
        }
        
        & .content {
            height: calc(100% - 3rem);
            padding: ${layout.standardGap};
            box-sizing: border-box;
            overflow-y: auto;
        }
    }
`;
