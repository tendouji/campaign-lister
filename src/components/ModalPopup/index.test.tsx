import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import ModalPopup from './';


describe('<ModalPopup />', () => {
    const props = {
        show: true,
        title: 'Test Title',
        content: 'Test Message',
    };

    it('renders without crashing with props', () => {
        const wrapper = shallow(<ModalPopup {...props} />);
        expect(wrapper.hasClass('show')).toEqual(true);
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it('move out of screen when close button is clicked', () => {
        const wrapper = shallow(<ModalPopup {...props} />);
        wrapper.setProps({
            close: () => wrapper.setProps({ show: false })
        });

        const closeBtn = wrapper.find('.close');
        closeBtn.simulate('click');

        expect(wrapper.hasClass('show')).toEqual(false);
    });
});