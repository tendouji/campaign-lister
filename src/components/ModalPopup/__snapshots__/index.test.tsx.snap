// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalPopup /> renders without crashing with props 1`] = `
<styled.div
  className="modal-popup show"
>
  <div
    className="bg"
  />
  <div
    className="fg"
  >
    <div
      className="header"
    >
      <div
        className="title"
      >
        Test Title
      </div>
      <div
        className="close"
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                352,
                512,
                Array [],
                "f00d",
                "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
              ],
              "iconName": "times",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </div>
    </div>
    <div
      className="content"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "Test Message",
          }
        }
      />
    </div>
  </div>
</styled.div>
`;
